contracts/modules/token/TokenController.sol

```diff
@@ -310,6 +308,70 @@ contract TokenController is ITokenController, LockHandler, MasterAwareV2 {
     }
   }
 
+  /// @notice Withdraws NXM from the Nexus platform based on specified options.
+  /// @dev Ensure the NXM is available and not locked before withdrawal. Only set flags in `WithdrawNxmOptions` for withdrawable NXM.
+  ///      Reverts if some of the NXM being withdrawn is locked or unavailable.
+  /// @param member Address of the member to withdraw NXM from.
+  /// @param stakingPoolDeposit Details for withdrawing from staking pools.
+  /// @param coverNotes Details for withdrawing V1 cover notes.
+  /// @param batchSize The maximum number of iterations to avoid unbounded loops when withdrawing governance and/or assessment rewards.
+  /// @param withdrawOptions Options specifying which NXM types to withdraw, set flags to true to include specific withdrawable NXM.
+  function withdrawNXM(
+    address member,
+    StakingPoolDeposit calldata stakingPoolDeposit,
+    V1CoverNotes calldata coverNotes,
+    uint batchSize,
+    WithdrawNxmOptions calldata withdrawOptions
+  ) external whenNotPaused {
+
+    // assessment stake
+    if (withdrawOptions.assessmentStake) {
+      IAssessment _assessment = assessment();
+      (uint96 amount, , ) = _assessment.stakeOf(member);
+      _assessment.unstakeFor(member, amount, member);
+    }
+
+    // assessment and governance rewards
+    if (withdrawOptions.governanceRewards || withdrawOptions.assessmentRewards) {
+      withdrawPendingRewards(
+        member,
+        withdrawOptions.governanceRewards,
+        withdrawOptions.assessmentRewards,
+        batchSize
+      );
+    }
+
+    // staking pools rewards and stake (including manager rewards)
+    if (withdrawOptions.stakingPoolStake || withdrawOptions.stakingPoolRewards) {
+      for (uint i = 0; i < stakingPoolDeposit.tokenIds.length; i++) {
+        uint tokenId = stakingPoolDeposit.tokenIds[i];
+        uint[] memory trancheIds = stakingPoolDeposit.tokenTrancheIds[i];
+        uint poolId = stakingNFT.stakingPoolOf(tokenId);
+        stakingPool(poolId).withdraw(tokenId, withdrawOptions.stakingPoolStake, withdrawOptions.stakingPoolRewards, trancheIds);
+      }
+    }
+
+    // v1 cover notes
```

Comment:
wild idea - we could simplify this a bit by either:
1. making V1 NXM always mandatory to withdraw 
2. have 1 flag for all different V1 NXM token sources

the goal is for users to withdraw as soon as possible all the V1 tokens

Reply:
These will now always be withdrawn (no risk of revert):
* `_withdrawClaimAssessmentTokensForUser`
* `pooledStaking.withdrawReward`

Theses ones still need to be validated by the FE as they can revert:
* `withdrawCoverNote` - reverts if user has no locked cover notes
* `pooledStaking.withdrawForUser` - reverts if user is ARMOR_STAKER | HUGH | NM_FOUNDATION

The FE should always try to withdraw v1 NXM and only skip if they will cause a revert.